core/sqf/src/tm/tmlib.cpp

```diff
@@ -647,6 +647,21 @@ short REGISTERREGION(long transid, long startid, int pv_port, char *pa_hostname,
    }
 
    TM_Transaction *lp_currTrans = gp_trans_thr->get_current();
+   // Check if the passed-in transid is known to the thread
+   // if so, make that Transid current
+   if (lp_currTrans == NULL) {
+      lp_currTrans = gp_trans_thr->get_trans (lv_transid.get_native_type());
+      if (lp_currTrans != NULL) 
+         gp_trans_thr->set_current(lp_currTrans);
+   } 
+   // Check if the thread's current transid matches the passed-in transid
+   // If not, check if the passed-in transid is known to the thread
+   // if so, make that Transid current
+   else if (! lp_currTrans->equal(lv_transid)) {
+      lp_currTrans = gp_trans_thr->get_trans (lv_transid.get_native_type());
+      if (lp_currTrans != NULL) 
+         gp_trans_thr->set_current(lp_currTrans);
+   }
    TM_Transseq_Type lv_savedStartId = gp_trans_thr->get_startid();
```

Comment:
Similar code exists few lines above and lv_savedStartId and lv_startId are one and the same at line 677. Also, can we take care of this if there are any issue in the next check-in